core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1020,6 +1052,87 @@ public void releaseRegisterLease(long workerId) {
     mRegisterLeaseManager.releaseLease(workerId);
   }
 
+  @Override
+  public String getClusterId() {
+    try {
+      // ClusterId is generated by MetaMaster,
+      // MetaMaster is constructed later than BlockMaster,
+      // so wait for MetaMaster to generate and set ClusterId.
+      CommonUtils.waitFor(this + " getClusterId",
+          () -> !mClusterId.get().equals(IdUtils.INVALID_CLUSTER_ID),
+          WaitForOptions.defaults().setTimeoutMs(5000));
+    } catch (Exception e) {
+      LOG.error("getClusterId {}: ", IdUtils.INVALID_CLUSTER_ID, e);
```

Comment:
When we cannot get a cluster ID, I think the master logic cannot continue and we should exit with a RuntimeException